### PR TITLE
Add documentation for the Daml Profiler

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin

- [Daml Profiler] Daml Connect EE now includes a profiler which can be
used to profile Daml execution time.

changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/tools/profiler.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+on that.
+
+Caveats
+=======
+
+1. The profiler currently does not take time into account that is
+   spent outside of pure interpretation, e.g., time needed to fetch a
+   contract from the database.
+
+2. The profiler operatos on Daml-LF. This means that the identifiers
```
Comment (bame-da) [inline docs/source/tools/profiler.rst:89]:
```suggestion
2. The profiler operates on Daml-LF. This means that the identifiers
```

File: daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala [hunk 2]
```diff
@@ -363,10 +363,9 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
           nodeSeeds = onLedger.ptx.nodeSeeds.toImmArray,
         )
         config.profileDir.foreach { dir =>
-          val hash = meta.nodeSeeds(0)._2.toHexString
           val desc = Engine.profileDesc(tx)
-          machine.profile.name = s"$desc-${hash.substring(0, 6)}"
-          val profileFile = dir.resolve(s"${meta.submissionTime}-$desc-$hash.json")
+          machine.profile.name = s"${meta.submissionTime}-$desc"
+          val profileFile = dir.resolve(s"${meta.submissionTime}-$desc.json")
```
Comment (cocreature) [inline daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala:368]: This change is deliberate. Including the transaction seed makes no sense for users.

**Review by remyhaemmerle-da — APPROVED** [review]
Nice.

**Pull request merged**